src/main.rs

```diff
@@ -187,48 +182,44 @@ fn get_linear_ticket(ticket_id: &Option<String>) -> Option<LinearIssue> {
     }
 }
 
-fn get_overview_str(commits: &Vec<Commit>) -> String {
-    let mut overview_str = String::new();
-
-    for (i, commit) in commits.iter().enumerate() {
-        let mut str = format!("- {}", commit.message);
-        if i < commits.len() - 1 {
-            str.push('\n');
-        }
-
-        overview_str.push_str(&str);
-    }
+fn get_overview_str(commits: &[Commit]) -> String {
+    let overview_str = commits
+        .iter()
+        .map(|commit| format!("- {}", commit.message))
+        .collect::<Vec<String>>()
+        .join("\n");
 
     overview_str
 }
 
 fn get_context_str(linear_ticket: &Option<LinearIssue>) -> String {
-    match &linear_ticket {
-        None => String::new(),
-        Some(ticket) => {
-            format!("{}\n\n{}", &ticket.url, &ticket.description)
-        }
-    }
+    linear_ticket
+        .as_ref()
+        .map(|ticket| format!("{}\n\n{}", ticket.url, ticket.description))
+        .unwrap_or_default()
 }
 
 // TODO: If no name has been generated, should open an editor for it / prompt for it
 fn get_pr_title(
     linear_ticket: &Option<LinearIssue>,
     linear_ticket_id: &Option<String>,
 ) -> Option<String> {
-    match &linear_ticket {
-        None => None,
-        Some(ticket) => {
+    if linear_ticket.is_none() {
+        Some("<!--- The title of your pull request. Save and close this file to continue. --->".to_string())
+    } else {
```

Comment:
Any particular reason to refactor away from `match`? I think this fixes an existing bug where you can't open a PR if there's no associated Linear ticket, but I prefer `match` since it requires exhaustive checking